xm-commons-security/src/main/java/com/icthh/xm/commons/security/oauth2/OAuth2JwtAccessTokenConverter.java

```diff
@@ -68,7 +68,9 @@ private boolean tryCreateSignatureVerifier() {
         try {
             SignatureVerifier verifier = signatureVerifierClient.getSignatureVerifier();
             if (verifier != null) {
-                setVerifier(verifier);
+                synchronized (signatureVerifierClient) {
+                    setVerifier(verifier);
+                };
                 lastKeyFetchTimestamp = t;
```

Comment:
I'm worried, why lastKeyFetchTimestamp = t; - outside synchronized block if this is write operation too